server/belga/io/feed_parsers/base_belga_newsml_1_2.py

```diff
@@ -555,9 +558,14 @@ def parser_descriptivemetadata(self, item, descript_el):
                 item['extra']['how_present'] = how_present_el
 
             elements = location_el.findall('Property')
+            # avoid repeatedly calling db while finding country
+            self.country = self._get_cv('country').get('items', [])
```

Comment:
`self.countries`

Reply:
I think this is not the correct place to preload countries. Theoretically, if `location_el is None` the `self.country` will not be filled, but if you call `_get_country` later it will raise an error because self.country is not even defined.
I would preload countries in the constructor (__init__) or even with lazy load like that:

```
def __init__(self):
    super().__init__()
    self._countries = []

 ....

def _get_country(self, country_code):

    if not self._countries:
        self._countries = self._get_cv('country').get('items', [])

    country_keyword = [
        c for c in self._countries
        if c.get('qcode') == 'country_' + country_code.lower() and c.get('is_active')
    ]
    # add scheme and remove is_active
    for c in country_keyword:
        c['scheme'] = 'country'
        if 'is_active' in c:
            c.pop('is_active')
    return country_keyword
```
⚠️ this snipped is just from my head, it's not tested for sure.

Reply:
Good point, make sense to do lazy for countries